pkg/demoinfocs/game_state.go

```diff
@@ -174,7 +175,15 @@ func (gs gameState) PlayerResourceEntity() st.Entity {
 	return gs.playerResourceEntity
 }
 
-func entityIDFromHandle(handle uint64) int {
+func entityIDFromHandle(handle uint64, isS2 bool) int {
+	if isS2 {
```

Comment:
I needed to distinguish Source 1 and 2 here since constants are different.
The easier but not nice solution I found is to inject a boolean here and as a side effect pass the `isSource2` function from `parser` to the struct `participants`.